### PR TITLE
feature 926 TCMPRPlotter improvements

### LINKED ISSUE #926: Improvements to TCMPRPlotter wrapper logging and output control
There are a few improvements that could be made to TCMPRPlotter:
* Currently TCMPRPlotter does not call the functionality to run commands that the rest of the wrappers do, so it does not output any of the log information from the Rscript run
* The environment variable(s) set by the wrapper and the command called are not output in the logs in the same way as the other wrappers. This may be remedied with the changes for the first item.
* The error handling logic references a produtil item that has not been imported, so the wrapper crashes on error

We should consider if time is better spent updating this wrapper or implement the R logic in Python inside METplotpy and replace this wrapper with a call to METplotpy.

## Describe the Enhancement ##
Enhance the TCMPRPlotter wrapper to implement the improvements listed above.

### Time Estimate ###
~3 days

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [X] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
4.1.0

### Funding Source ###
2788881

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [X] Select **component(s)**
- [X] Select **priority**
- [X] Select **requestor(s)**

### Projects and Milestone ###
- [X] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [X] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
If we decide to migrate this logic to METplotpy/METcalcpy instead of fixing the wrapper, we will need to create issues in the appropriate repositories:
- [ ] [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [x] Complete the development and test your changes.
- [x] Add/update log messages for easier debugging.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [x] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [x] Delete your fork or branch.
- [x] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
Changes include:

* Fixed crash when R script command failed
* Refactored logic to build commands
* Loop over -dep and -plot list and run once for each to allow customization of each plot (title, filename, labels, etc.)
* Added variables to set label string for -dep and -plot values to be used in title
* Added unit tests for wrapper
* Cleaned up conf file for TCMPRPlotter basic use case

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Code from feature branch was installed on hera and jet for testing by end users. The changes allowed them to generate the plots they need.

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review code changes, ensure all automated tests pass.

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [X] Please complete this pull request review by **6/24/2021**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [x] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [x] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Users_Guide/glossary.rst [hunk 1]
```diff
@@ -3327,7 +3352,15 @@ METplus Configuration Glossary
      .. warning:: **DEPRECATED:** Please use :term:`TCMPR_PLOTTER_PLOT_TYPES` instead.
 
    TCMPR_PLOTTER_PLOT_TYPES
-     Specify what plot types are desired for the TC Matched Pairs plotting tool. By default, a boxplot is generated if this is undefined in the configuration file. If other plots are requested and a boxplot is also desired, you must explicitly listboxplot in your list of plot types. Supported plot types: BOXPLOT, POINT, MEAN, MEDIAN, RELPERF (relative performance), RANK (time series of ranks for the first model), SCATTER, SKILL_MN (mean skill scores) and SKILL_MD (median skill scores).
+     Specify what plot types are desired for the TC Matched Pairs plotting
+     tool. By default, a boxplot is generated if this is undefined in the
+     configuration file. If other plots are requested and a boxplot is also
+     desired, you must explicitly listboxplot in your list of plot types.
```
Comment (bikegeek) [inline docs/Users_Guide/glossary.rst]: I think there is a typo (from the original):

_desired, you must explicitly **listboxplot** in your list of plot types_ 

should be
_desired, you must explicitly **list boxplot** in your list of plot types_

Comment (georgemccabe) [reply docs/Users_Guide/glossary.rst]: Thanks, fixed it.

**Comment by georgemccabe** [comment]
Note that a change in MET caused the GenVxMask use case to fail. This can be ignored for this pull request. It is being address in #960

**Review by bikegeek — APPROVED** [review]
Looks good.  I approve

**Comment by georgemccabe** [comment]
Confirmed the only failed test was due to the GenVxMask issue that was resolved in another pull request. Merging these changes now.

**Pull request merged**